mypy/newsemanal/semanal.py

```diff
@@ -3704,8 +3614,9 @@ def lookup_qualified(self, name: str, ctx: Context,
                             gvar = self.create_getattr_var(names['__getattr__'],
                                                            parts[i], parts[i])
                             if gvar:
-                                names[name] = gvar
-                                n = gvar
+                                n = SymbolTableNode(GDEF, gvar)
+                                # TODO: What was the purpose of this?
+                                # names[name] = n
```

Comment:
TBH, I forgot why. Approximately the idea was that two expressions like `a.A` (if `a` is a module with `__getattr__`) should point to the same "auto-generated" node. Probably I added this just in case to avoid troubles in incremental modes. Also this still sounds consistent to me.